app/models/event_procedure.rb

```diff
@@ -32,10 +32,19 @@ class EventProcedure < ApplicationRecord
   validates :urgency, inclusion: [true, false], if: -> { health_insurance? }
   validates :payment, presence: true
 
+  validate :match_user_with_patient_user
   validate :custom_and_urgency_cannot_be_true
 
   private
 
+  def match_user_with_patient_user
+    return if patient.nil? || user.nil?
+
+    return unless patient.user_id != user_id
```

Comment:
Avoid using `unless` together with negation 
```suggestion
    return if patient.user_id == user_id
```